src/transactions/BumpSequenceOpFrame.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/BumpSequenceOpFrame.h"
+#include "transactions/TransactionFrame.h"
+#include "crypto/SignerKey.h"
+#include "database/Database.h"
+#include "main/Application.h"
+#include "medida/meter.h"
+#include "medida/metrics_registry.h"
+
+namespace stellar
+{
+using xdr::operator==;
+
+
+BumpSequenceOpFrame::BumpSequenceOpFrame(Operation const& op, OperationResult& res,
+                                     TransactionFrame& parentTx)
+    : OperationFrame(op, res, parentTx)
+    , mBumpSequence(mOperation.body.bumpSequenceOp())
+{
+}
+
+ThresholdLevel
+BumpSequenceOpFrame::getThresholdLevel() const
+{
+    // bumping sequence is low threshold
+    return ThresholdLevel::LOW;
+}
+
+bool
+BumpSequenceOpFrame::doApply(Application& app, LedgerDelta& delta,
+                           LedgerManager& ledgerManager)
+{
+    Database& db = ledgerManager.getDatabase();
+
+    AccountFrame& bumpAccount = getSourceAccount();
```

Comment:
Just a lurker here and not a stellar-core expert so ignore if I'm wrong but shouldn't you read the account from DB like:
```cpp
AccountFrame& bumpAccount = AccountFrame::loadAccount(delta, mSourceAccount->getID(), db);
```
Current code will ignore the previous bumps within the same transaction.

Reply:
I had something like this previously, but I thought it was unnecessary, but certainly loading this way is the safer option.

Can someone else comment on this one?

Reply:
My understanding is that mSourceAccount is a shared_ptr (presumably s.t. the invariant one shared_ptr per AccountID?), so it should be fine.